appservice/src/SiteWrapper.ts

```diff
@@ -143,11 +208,27 @@ export class SiteWrapper {
     private async getKuduClient(client: WebSiteManagementClient): Promise<KuduClient> {
         const user: User = await this.getWebAppPublishCredential(client);
         if (!user.publishingUserName || !user.publishingPassword) {
-            throw new ArgumentError(user);
+            throw new errors.ArgumentError(user);
         }
 
         const cred: BasicAuthenticationCredentials = new BasicAuthenticationCredentials(user.publishingUserName, user.publishingPassword);
 
         return new KuduClient(cred, `https://${this.appName}.scm.azurewebsites.net`);
     }
+
+    private async updateScmType(client: WebSiteManagementClient, config: SiteConfigResource, scmType: string): Promise<string | undefined> {
+        const oldScmType: string = config.scmType;
+        const updateScm: string = localize('updateScm', `Deployment source for "${this.appName}" is set as "${oldScmType}".  Change to "${scmType}"?`);
+        const yes: string = 'Yes';
+        let input: string | undefined;
+
+        const updateConfig: SiteConfigResource = config;
+        updateConfig.scmType = scmType;
+        // to update one property, a complete config file must be sent
+        input = await vscode.window.showWarningMessage(updateScm, yes);
+        if (input === 'Yes') {
+            await this.updateConfiguration(client, updateConfig);
```

Comment:
For this function I would return 
```
const newConfig = await this.updateConfiguration(client, updateConfig);
return newConfig.scmType;
```
Otherwise return undefined